subprojects/docs/src/samples/testing/junitplatform/engine/build.gradle

```diff
@@ -0,0 +1,17 @@
+apply plugin: 'java'
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.0.3'
+    testImplementation 'org.junit.vintage:junit-vintage-engine:4.12.3'
```

Comment:
This should be `testRuntimeOnly`, just like for the Jupiter engine. You'll also need `testImplementation 'junit:junit:4.12'`.

Reply:
You mean we use `junit:junit:4.12` to compile the tests and `vintage-engine` to run the tests? Thanks for reminding.

Reply:
What is the benefit of compiling and running with different dependencies? I would have thought that it would be simpler, and potentially less error-prone, to compile and run with the same.

Reply:
Well, your tests should not have a compile-time dependency on any class in `junit-vintage-engine`. Thus, even though adding it to `testImplementation` works, the JUnit teams' recommended way to do dependency management is to only add the engine to the runtime classpath so you don't accidentally depend on code in `junit-vintage-engine`.